src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp

```diff
@@ -1043,6 +1055,80 @@ struct npc_darkfallen_tactician : public DarkFallenAI
     }
 };
 
+struct npc_nerubar_broodkeeper : public ScriptedAI
+{
+    npc_nerubar_broodkeeper(Creature* creature) : ScriptedAI(creature)
+    {
+        Initialize();
+    }
+
+    void Initialize()
+    {
+        CastNet = false;
+    }
+
+    void Reset() override
+    {
+        events.Reset();
+        events.ScheduleEvent(EVENT_CRYPT_SCARABS, 1s, 3s);  // Crypt Scarabs
+        events.ScheduleEvent(EVENT_DARK_MENDING, 10s, 15s); // Dark Mending
+        events.ScheduleEvent(EVENT_WEB_WRAP, 6s, 8s);       // Web Wrap
+    }
+
+    void SetData(uint32 type, uint32 data) override
+    {
+        if (type == 1 && data == 1)
+        {
+            me->GetMotionMaster()->MoveFall();
+            me->SetHover(false);
+            me->SetByteValue(UNIT_FIELD_BYTES_1, UNIT_BYTES_1_OFFSET_ANIM_TIER, 0);
+            if (!CastNet)
+            {
+                CastNet = true;
+                me->RemoveAllAuras();
+                DoCastSelf(SPELL_CAST_NET_STARTROOM, true);
+            }
+        }
+    }
+
+    void UpdateAI(uint32 diff) override
+    {
+        if (!UpdateVictim())
+            return;
+
+        events.Update(diff);
+
+        switch (events.ExecuteEvent())
+        {
+            case EVENT_CRYPT_SCARABS:
+                me->CastSpell(me->GetVictim(), SPELL_CRYPT_SCARABS, false);
+                events.Repeat(3s, 5s);
+                break;
+            case EVENT_DARK_MENDING:
+                if (Unit* target = DoSelectLowestHpFriendly(30.0f, 100000))
+                {
+                    if (me->GetInstanceScript() && (me->GetMap()->GetSpawnMode() & 0 || me->GetMap()->GetSpawnMode() & 1))
```

Comment:
I'm curious about what you are trying to accomplish with ```me->GetMap()->GetSpawnMode() & 0```

Reply:
> I'm curious about what you are trying to accomplish with `me->GetMap()->GetSpawnMode() & 0`

I wanted to give the difference of heal in heroics and normality.

Reply:
but ```me->GetMap()->GetSpawnMode() & 0``` is not really "valid" C++ code

Reply:
Done